lib/thor/parser/options.rb

```diff
@@ -83,7 +84,11 @@ def parse(args) # rubocop:disable MethodLength
             when SHORT_SQ_RE
               unshift($1.split("").map { |f| "-#{f}" })
               next
-            when EQ_RE, SHORT_NUM
+            when EQ_RE
+              unshift($2)
+              switch = $1
+              @force_value = true
```

Comment:
I think setting this state here will broke it if there are more options after the first with `-` in the value, can you add tests to make sure it will work?